### ISSUE TITLE
Poetry does not respect the include in the pyproject.toml when the listed file is in the `.gitignore`

### ISSUE THREAD

**Issue by johnwalz97** [python-poetry/poetry#4558]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: 
- **Poetry version**: 
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: 

## Issue

I am trying to include in the published package, some files that are build artifacts so are in the `.gitignore`. I am really struggling to get everything to work, because no matter what I do Poetry seems to accept the `.gitignore` over the globs listed in the `pyproject.toml` `tool.poetry.include` field. I ran it with `-vvv` and Poetry is straight up not including the files. However, when I remove the pattern for the files from the `.gitignore` it works fine. I am going to try and use a different version of Poetry but I think this is a important bug that should be fixed. Thanks!

**Comment by johnwalz97** [comment]
This is my `pyproject.toml` for reference.

```
[tool.poetry]
name = "jupyterlab_toniq"
version = "1.0.0-alpha.5"
description = "JupyterLab Extension for Toniq Widgets and Customizations"
authors = ["{redacted}"]
include = ["jupyterlab/labextension/**"]

[[tool.poetry.source]]
name = "toniq"
url = "https://{redacted}/simple/"

[tool.poetry.dependencies]
python = "^3.7.1"
jupyter-server = "^1.11.0"

[tool.poetry.dev-dependencies]
black = {version = "^21.9b0", allow-prereleases = true}
jupyterlab = "^3.1.12"
click = "^8.0.1"
pylint = "^2.11.1"
toniq = ">=1.30.0"

[build-system]
requires = ["poetry-core>=1.0.0"]
build-backend = "poetry.core.masonry.api"
```

**Comment by johnwalz97** [comment]
I tried on Poetry 1.8.0. and Poetry 1.10.0 and neither worked.

**Comment by johnwalz97** [comment]
I am using a Makefile to run the Poetry commands just because people at my company are not very familiar with Poetry and its use and its sometimes easier to have clearly defined, project-specific commands. As a workaround for now, I just added a couple lines to the make target that calls `poetry build` which temporarily move the `.gitignore` to `.gitignore.tmp` and then back once the build is complete...

**Comment by finswimmer** [comment]
Hello @johnwalz97,

I think your wildcard usage is wrong. Try this:

```py
include = ["jupyterlab/labextension/**/*"]
```

fin swimmer

**Comment by johnwalz97** [comment]
Hey @finswimmer. Thanks for the response!

You are correct that was my issue. Well, that and a typo. Initially I had used the correct wildcard pattern but I mis-typed the directory name. Then I realized that when I removed the directory from the `.gitignore` it worked. So I assumed something was up with Poetry. The reason that worked though was just because the directory was inside the Python package directory so it was automatically included even if the wildcard was off...

I was doing some late-night coding and was struggling with a lot of other issues not related to Poetry. This was my first issue on GitHub so I feel pretty stupid lol. Thank you so much though for the fix!

- John
